src/peer_digest.cc

```diff
@@ -717,7 +668,7 @@ peerDigestFetchedEnough(DigestFetchState * fetch, char *buf, ssize_t size, const
     /* continue checking (with pd and host known and valid) */
 
     if (!reason) {
-        if (!cbdataReferenceValid(pd->peer))
+        if (!pd->peer)
```

Comment:
This is another (potential) bug fix in this PR AFAICT: The official code does not interpret nil `pd->peer` as a `reason`-setting problem[^1], but should. Again, I adjusted PR description to disclose these fixes.

[^1]: cbdataReferenceValid(nullptr) returns true